vnext/ReactWindowsCore/WinRTWebSocketResource.h

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// clang-format off
+
+#pragma once
+
+//#include <Mso/eventWaitHandle/eventWaitHandle.h>
+#include <winrt/Windows.Networking.Sockets.h>
+#include <winrt/Windows.Storage.Streams.h>
+#include <IWebSocketResource.h>
+
+#include <future>
+#include <queue>
+
+namespace Microsoft::React
+{
+
+class WinRTWebSocketResource : public IWebSocketResource
+{
+  winrt::Windows::Foundation::Uri m_uri;
+  winrt::Windows::Networking::Sockets::MessageWebSocket m_socket;
+  winrt::Windows::Networking::Sockets::MessageWebSocket::MessageReceived_revoker m_revoker;
+  winrt::Windows::Storage::Streams::DataWriter m_writer;
+
+  std::atomic_bool m_connectRequested;
+  std::atomic_bool m_sendRequested;
+  // Mso::ManualResetEvent m_connectPerformed;
+  // Mso::ManualResetEvent m_sendPerformed;
+  std::promise<void> m_connectPerformed;
+
+  CloseCode m_closeCode;
+  std::string m_closeReason;
+  std::queue<std::pair<std::string, bool>> m_writeQueue;
+
+  std::function<void()> m_connectHandler;
+  std::function<void()> m_pingHandler;
+  std::function<void(std::size_t)> m_writeHandler;
+  std::function<void(std::size_t, const std::string&)> m_readHandler;
+  std::function<void(CloseCode, const std::string&)> m_closeHandler;
+  std::function<void(Error&&)> m_errorHandler;
+
+  WinRTWebSocketResource(winrt::Windows::Foundation::Uri&& uri);
+  ~WinRTWebSocketResource() override;
+
+  winrt::fire_and_forget PerformConnect();
+  winrt::fire_and_forget PerformPing();
+  winrt::fire_and_forget PerformWrite();
+  winrt::fire_and_forget PerformClose();
+
+  void OnMessageReceived(
+    winrt::Windows::Networking::Sockets::IWebSocket const& sender,
+    winrt::Windows::Networking::Sockets::MessageWebSocketMessageReceivedEventArgs const& args);
+  void OnClosed(
+    winrt::Windows::Networking::Sockets::IWebSocket const& sender,
+    winrt::Windows::Networking::Sockets::WebSocketClosedEventArgs const& args);
+
+  void Stop();
+
+public:
+  WinRTWebSocketResource(const std::string& urlString);
+
+#pragma region IWebSocketResource
+
+  /// <summary>
```

Comment:
Nit: after a recent poll, people liked either plain comments or Javadoc style Doxygen for RNW.

Reply:
Sure. I copied that chunk verbosely fro m BeastWSRsc. I'll fully document the class with Doxygen style in the next PR.

Reply:
This comment style is ok too. As long as IDEs such as Visual Studios display it properly

---
In reply to: [394799507](https://github.com/microsoft/react-native-windows/pull/4330#discussion_r394799507) [](ancestors = 394799507)